power_grid_model_io_native_c/power_grid_model_io_native/include/power_grid_model_io_native/pgm_vnf_converter/vnf_parser.hpp

```diff
@@ -0,0 +1,100 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#pragma once
+#ifndef POWER_GRID_MODEL_IO_NATIVE_C_VNF_PARSER_HPP
+#define POWER_GRID_MODEL_IO_NATIVE_C_VNF_PARSER_HPP
+
+#include <power_grid_model_io_native/common/common.hpp>
+
+#include <power_grid_model/auxiliary/input.hpp>
+#include <power_grid_model/container.hpp>
+
+#include <iostream>
+#include <map>
+#include <regex>
+
+namespace power_grid_model_io_native {
+using VisionGUID = std::string;
+
+struct VnfNode {
+    VisionGUID guid{"nan"};
+    double u_nom{nan};
+};
+
+using VnfGrid = pgm::Container<VnfNode>;
+template <typename Identifier> using IdentifierLookup = std::unordered_map<Identifier, ID>;
+using VisionGUIDLookup = IdentifierLookup<VisionGUID>;
+
+class PgmVnfParser {
+  public:
+    PgmVnfParser(std::string_view vnf_data);
```

Comment:
cfr. sonar cloud: https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=26&id=PowerGridModel_power-grid-model-io-native&open=AZSn72zt5oT0QmyLmKio&tab=why

```suggestion
    explicit(false) PgmVnfParser(std::string_view vnf_data);
```